### ISSUE TITLE
Invalid TS4029: Need TS 2.7

### ISSUE THREAD

**Issue by Vandivier** [angular/angular#21157]
This issue is solved in TS 2.7, [see here](https://github.com/Microsoft/TypeScript/issues/20657).

But, if I bump to 2.7 Angular throws that my version is too advanced. Is there really any issue with using 2.7?
![image](https://user-images.githubusercontent.com/5559355/34316927-566154da-e770-11e7-92db-ce7eb45571df.png)

I see that error, TS4029, in the following service. It throws on 'obsArroArticles =...' using Angular 5.1.1 / ts 2.4.2

```
import { Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs/BehaviorSubject';

@Injectable()

export class ServiceArticlesService {

  private arroArticles = new BehaviorSubject<any>([{ // TODO: Article Type, not any
      'sContent': 'lorem ipsum, etc, etc.',
      'sTitle': 'This is my first article!'
  }, {
      'sContent': 'scoobie doobie do. where r u.',
      'sTitle': 'This is my second [:'
  }]);

  obsArroArticles = this.arroArticles.asObservable();

  constructor() { }

  fChangeArticles(_oArticle) {
      this.arroArticles.next(_oArticle);
  }
}

```

**Comment by ericmartinezr** [comment]
They just merged the support for ts2.6 https://github.com/angular/angular/commit/83d207d0a795bc1945c4661cb5ab73d2e671c1ce

**Comment by Vandivier** [comment]
quite aware. and this issue will remain outstanding afaik until 2.7

**Comment by IgorMinar** [comment]
We have no idea if 2.7-dev is compatible or not.

Typically every *minor* version of typescript changes the AST apis which break our compiler so we need to tweak things every time.

The errors above are in fact just warnings, so you can try 2.7 for yourself. Also feel free to fork Angular and test 2.7.

**Comment by mhevery** [comment]
closing since there is no action on our part.

**Comment by mlc-mlapis** [comment]
@Vandivier ... you don't need TS 2.7 ... it is solved in supported TS 2.5.3 by adding:

`import { Observable } from 'rxjs/Observable';`
